samples/core/src/main/java/io/oasp/gastronomy/restaurant/salesmanagement/dataaccess/impl/dao/OrderPositionDaoImpl.java

```diff
@@ -36,10 +37,10 @@
   public List<OrderPositionEntity> findOrderPositionsByOrder(Long orderId) {
 
     OrderPositionEntity orderPosition = Alias.alias(OrderPositionEntity.class);
-    EntityPathBase<OrderPositionEntity> alias = Alias.$(orderPosition);
-    JPAQuery query = new JPAQuery(getEntityManager()).from(alias);
-    query.where(Alias.$(orderPosition.getOrder().getId()).eq(orderId));
-    return query.list(alias);
+    EntityPathBase<OrderPositionEntity> alias = $(orderPosition);
+    JPAQuery<OrderPositionEntity> query = new JPAQuery<OrderPositionEntity>(getEntityManager()).from(alias);
```

Comment:
duplicate type parameter declaration as also in other Dao's multiple times.
Use diamond operator.

Reply:
Yes, seems that my eclipse is misconfigured (in some cases it is also a java7 type inference issue but surely not here). I will fix this.

Reply:
Actually the entire restaurant sample is going to be deleted very soon. We should not make any fuss about this code. I just invest what is necessary to not break the build. See PR #629 - if that has been merged, we can start PR to delete restaurant.

Reply:
Actually you are even wrong here. In this case the Diamond Operator can not be used. This would lead to a compiler error. Yes, Java is sometimes sick.